src/lua/loaders.lua

```diff
@@ -250,6 +250,47 @@ table.insert(package.loaders, 5, gen_loader_func(search_rocks_lib, load_lib))
 -- package.cpath  7
 -- croot          8
 
+-- Search for modules next to the main script.
+local script = minifio.script()
+if script ~= nil and script ~= '-' then
```

Comment:
Side note: This part gave me some interesting ideas to explore:

```sh
$ cat $PWD/-
print("test")
$ luajit -e 'print("-e")' -- -
-e
test
$ lua -e 'print("-e")' -- -
-e
test
$ src/tarantool -e 'print("-e")' -- -
-e
^C^C^C^C^C^C^Z
[1]+  Stopped                 src/tarantool -e 'print("-e")' -- -
```

`--`: stops handling options in Lua/LuaJIT (see [Lua 5.1 Reference Manual](https://www.lua.org/manual/5.1/manual.html#6))
As far as it is not implemented via Tarantool we can ignore this case, for now.

Reply:
No problems.

```diff
diff --git a/src/lua/loaders.lua b/src/lua/loaders.lua
index 2d7cb69b0..794f74eb5 100644
--- a/src/lua/loaders.lua
+++ b/src/lua/loaders.lua
@@ -251,6 +251,10 @@ table.insert(package.loaders, 5, gen_loader_func(search_rocks_lib, load_lib))
 -- croot          8
 
 -- Search for modules next to the main script.
+--
+-- If the script is not provided (script == nil) or provided as
+-- stdin (script == '-'), there is no script directory, so nothing
+-- to do here.
 local script = minifio.script()
 if script ~= nil and script ~= '-' then
     -- It is important to obtain the directory at initialization,
```